addons/test_mail/tests/test_message_compose.py

```diff
@@ -365,3 +365,19 @@ def test_message_compose_portal_ok(self):
 
             self.assertEqual(self.test_record.message_ids[0].body, '<p>Body text 2</p>')
             self.assertEqual(self.test_record.message_ids[0].author_id, portal_user.partner_id)
+
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_auto_delete_sent_mail(self):
+        composer = self.env['mail.compose.message'].with_context({
+            'default_model': self.test_record._name,
+            'default_res_id': self.test_record.id,
+        }).sudo(self.user_employee).create({
+            'body': '<p>Test Body</p>',
+            'partner_ids': [(4, self.partner_1.id), (4, self.partner_2.id)]
+        })
+        composer.send_mail()
+
+        message = self.test_record.message_ids[0]
+
+        self.assertFalse(self.env['mail.mail'].search([('mail_message_id', '=', message.id)]),
```

Comment:
To be complete, you should test that a template with auto_delete set to True or False correctly keeps / removes notifications. That way we have a complete coverage.